ICRCs/ICRC-97/ICRC-97.md

```diff
@@ -0,0 +1,101 @@
+pairs. Where `Value` is recursive as defined in the ICRC-3 standard.
+
+ICRC-7 does not specify the representation of token metadata any further than that it is represented in a generic manner
+as a vector of (text, Value)-pairs. The ICRC-97 standard defines a token metadata
+standard for ICRC-7 and other NFT standards.
+
+### Entrypoint
+
+The following metadata property MUST be defined in the root of the token metadata if we want to return on-chain
+metadata.
+
+| Root Metadata Property | ICRC-3 Type     | Description                            |
+|------------------------|-----------------|----------------------------------------|
+| icrc97:metadata        | variant { Map } | Contains on-chain metadata properties. |
+
+Alternatively the following metadata property MUST be defined in the root of the token metadata if we want to return
+external
+off-chain metadata instead of on-chain metadata.
+
+| Root Metadata Property   | ICRC-3 Type     | Description                                       |
+|--------------------------|-----------------|---------------------------------------------------|
+| icrc97:external_metadata | variant { Map } | Contains below external JSON metadata properties. |
+
+| External Metadata Property | Optional | ICRC-3 Type      | Description                                                                    |
+|----------------------------|----------|------------------|--------------------------------------------------------------------------------|
+| url                        | No       | variant { Text } | URL that returns the metadata in JSON format (protocol is not limited to HTTP) |
+| sha256_hash                | Yes      | variant { Blob } | SHA-256 hash of HTTP response body bytes returned from above url.              |
+
+### Metadata properties
+
+Below table lists all (optional) metadata properties defined in this standard. Each property is defined in both ICRC-3
+and JSON type to support both on-chain `Value` and off-chain JSON metadata, respectively.
+
+| Metadata Property | Optional | ICRC-3 Type                             | JSON Type | Description                                             |
+|-------------------|----------|-----------------------------------------|-----------|---------------------------------------------------------|
+| external_url      | Yes      | variant { Text }                        | string    | URL that allows the user to view the item on your site. |
+| name              | Yes      | variant { Text }                        | string    | Plain text.                                             |
+| description       | Yes      | variant { Text }                        | string    | Markdown.                                               |
+| assets            | Yes      | variant { Array = vec variant { Map } } | object[]  | List of assets ordered by priority descending.          |
```

Comment:
> List of assets ordered by priority descending.

The order of the list of assets matters, dapps are expected to prioritize assets higher in the list.

Reply:
so as a consumer (e.g. explorer or marketplace) I would use the first occurrence of an asset with type image (or preview if existing), right? assuming that the NFT has an image attached.

if we can have more than one asset with purpose `irc97:preview`, I'd expect that `irc97:preview` differs from its corresponding `irc97:image`. otherwise the consumer could just resize the `irc97:image` to the needs.

Reply:
> I would use the first occurrence of an asset with type image (or preview if existing), right?

Yes, I'll clarify this in the spec a bit further.

> otherwise the consumer could just resize the irc97:image to the needs

I'll also clarify in the spec that `irc97:image` should be used if `irc97:preview` is missing.